derex/runner/project.py

```diff
@@ -51,6 +62,61 @@ class Project:
     # Path to a local docker-compose.yml file, if present
     local_compose: Optional[Path] = None
 
+    @property
+    def runmode(self) -> ProjectRunMode:
+        """The run mode of this project, either debug or production.
+        In debug mode django's runserver is used. Templates are reloaded
+        on every request and assets do not need to be collected.
+        In production mode gunicorn is run, and assets need to be compiled and collected.
+        """
+        name = "runmode"
+        mode_str = self._get_status(name)
+        if mode_str is not None:
+            if mode_str in ProjectRunMode.__members__:
+                return ProjectRunMode[mode_str]
+            # We found a string but we don't recognize it: warn the user
+            logger.warn(
+                f"Value `{mode_str}` found in `{self._status_filepath(name)}` "
+                "is not valid for runmode "
+                "(valid values are `debug` and `production`)"
+            )
+        default = self.config.get(f"default_{name}")
+        if default:
+            if default not in ProjectRunMode.__members__:
+                logger.warn(
+                    f"Value `{default}` found in config `{self.root / CONF_FILENAME}` "
+                    "is not a valid default for runmode "
+                    "(valid values are `debug` and `production`)"
+                )
+            else:
+                return ProjectRunMode[default]
+        return next(iter(ProjectRunMode))  # Return the first by default
+
+    @runmode.setter
+    def runmode(self, value: ProjectRunMode):
+        self._set_status("runmode", value.name)
+
+    def _get_status(self, name: str) -> Optional[str]:
+        """Read value for the desired status from the project directory.
+        """
+        filepath = self._status_filepath(name)
+        if filepath.exists():
+            return filepath.read_text()
+        return None
+
+    def _set_status(self, name: str, value: str):
+        """Persist a status in the project directory.
+        Each status will be written to a different file.
+        """
+        if not self._status_filepath(name).parent.exists():
+            self._status_filepath(name).parent.mkdir()
+        self._status_filepath(name).write_text(value)
```

Comment:
This append a newline on every write. [Looks like there is still no way to tell pathlib not to do that](https://bugs.python.org/issue23706).
Not a big issue but maybe we should consider rewriting the whole file.

Reply:
I don't understand. Can you provide an example of the behaviour you describe?

I see no newlines in the file (as expected) and the linked issue talks about newline styles, i.e. what specific bytes will be written to disk when we pass a string containing a newline to `write_text`. Since the string we pass has no newlines in it I miss how the linked issue can be relevant. Am I missing something?

Reply:
Nevermind, I don't know what I saw.